### PR TITLE
circuit: Make `TxInputNote` fields public

### PR DESCRIPTION
**Pull request by moCello**
Expose the fields of `TxInputNote` so it can also be constructed from the internal fields, without the sender-sk.

### CODE REVIEW

File: circuits/src/transaction.rs [hunk 1]
```diff
@@ -408,9 +263,44 @@ impl<const H: usize, const I: usize> TxCircuit<H, I> {
             sender_blinder,
         }
     }
+
+    /// Return the public-input values used for the circuit
+    pub fn public_inputs(&self) -> Vec<BlsScalar> {
```
Comment (xevisalle) [inline circuits/src/transaction.rs]: why is this needed if that's returned by the prover?

File: circuits/src/transaction.rs [hunk 2]
```diff
@@ -425,18 +315,132 @@ impl<const H: usize, const I: usize> Circuit for TxCircuit<H, I> {
...
-        nullify_gadget::<H, I>(
-            composer,
-            &payload_hash,
-            &self.root,
-            &self.tx_input_notes,
-            &self.tx_output_notes,
-            self.max_fee,
-            self.deposit,
-        )?;
+        // Append the root as public input
```
Comment (xevisalle) [inline circuits/src/transaction.rs:299]: which is the advantage of adding this here and changing the already peer-reviewed comments?

File: circuits/src/transaction.rs (hunk 1 above)
Comment (moCello) [reply circuits/src/transaction.rs]: because in practice the verifier doesn't have access to the data that the prover returns.
To be fair, I'm also not sure if the verifier will use this function downstream or if the pi will be generated from the transaction itself. I actually think the latter is the case.

File: circuits/src/transaction.rs (hunk 2 above)
Comment (moCello) [reply circuits/src/transaction.rs:299]: Ease of work: the other file was a real pain to debug and I changed only the comments that directly refer to the steps 1 - 6 in the documentation of the function.
I also thought that it is good to have the circuit implementation be more close to the circuit outlined in the specs with clear markers where what things are checked (e.g. membership, ownership, etc)

File: circuits/src/transaction.rs (hunk 1 above)
Comment (moCello) [reply circuits/src/transaction.rs]: This got me thinking, I'll remove the function again!

**Review by ureeves — APPROVED** [review]
LGTM!
